pkg/konfluxgen/dockerfile-component.template.yaml

```diff
@@ -3,7 +3,7 @@ kind: Component
 metadata:
   annotations:
     build.appstudio.openshift.io/pipeline: '{"name":"docker-build","bundle":"latest"}'
-  name: {{ truncate ( sanitize ( print .ProjectDirectoryImageBuildStepConfiguration.To "-" .ReleaseBuildConfiguration.Metadata.Branch ) ) }}
+  name: {{ sanitize .ComponentName }}
```

Comment:
nit: sanitize here probably not needed, as we sanitze it before we pass it to the template. Same probably for line 8 (application name)

Reply:
Yeah sure, but it's more robust